src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -181,6 +194,14 @@ public override void Initialize(AnalysisContext analysisContext)
                                 {
                                     rule = PureMethodRule;
                                 }
+                                else if (disposableType != null &&
+                                    targetMethod.ReturnType.ImplementsIDisposable(disposableType) &&
+                                    // We want to ignore some of the async methods from assertion frameworks
```

Comment:
I had to add those exclusions as one of the condition of the `ShouldSkipAnalyzing` isn't true (see later comment).